storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
Ok, I added `err` to all the returns where I could. But the ones right here were actually not naked returns, but returns from a closure without any return value.